### PR TITLE
Issue #3403765 by SV: Prevent unexpected error when creating content assigned to a group

### PR DESCRIPTION
**Pull request by volodymyr-sydor**
## Problem

When I tried to create an event and got an error:
Screenshot 2023-11-23 at 16 27 57

## Solution

Add a check to prevent an issue on event create page

## Issue tracker

- https://www.drupal.org/project/social/issues/3403765
- https://getopensocial.atlassian.net/browse/PROD-27295

## Theme issue tracker
N/A

## How to test

- [ ] Using the latest version of Open Social with the entity_access_by_field module enabled
- [ ] As a LU
- [ ] Try to create event that is assigned to a group with "group member" visibility
- [ ] When saving I expect to create a new event but instead see WSOD
- [ ] The expected result is attained when repeating the steps with this fix applied.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots

N/A

## Release notes

Add a check to prevent possible errors when creating content assigned to a specific group.

## Translations

N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

File: modules/custom/entity_access_by_field/src/EntityAccessHelper.php [hunk 1]
```diff
@@ -133,6 +133,7 @@ public static function entityAccessCheck(
             return self::NEUTRAL;
           }
           elseif (
+            !$entity->isNew() &&
```
Comment (ribel) [inline modules/custom/entity_access_by_field/src/EntityAccessHelper.php:136]: @volodymyr-sydor Could you explain more why do we need this check?
How we can make sure that it doesn't cause any regression for cases with both new and not new entities?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

File: modules/custom/entity_access_by_field/src/EntityAccessHelper.php (hunk 1 above)
Comment (volodymyr-sydor) [reply modules/custom/entity_access_by_field/src/EntityAccessHelper.php:136]: Currently we have: "Unexpected error when creating event assigned to a specific group"

So, during the creation a new entity there is called the current `entityAccessCheck` where we have:

`$group_contents = GroupContent::loadByEntity($entity);` 

but a new entity doesn't have an ID yet which caused that error, that's why we should prevent such cases

Also, this check we have for `getEntityAccessResult()` but for `entityAccessCheck()` method is missed:

    public static function getEntityAccessResult(
      EntityInterface $entity,
      string $operation,
      AccountInterface $account,
      string $permission = NULL
    ): AccessResultInterface {
      $access = self::entityAccessCheck(

     ....

    // If the social_event_invite module is enabled and a person got invited
    // then allow access to view the node.
    // @todo Come up with a better solution for this code.
    if (
      \Drupal::moduleHandler()->moduleExists('social_event_invite') &&
      $entity->getEntityTypeId() === 'node' &&
      !$entity->isNew() &&
      $operation === 'view'
    ) {

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3618-k2ejv4mmlyqvepx1eerz7zmtwe3mxmx0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/655f63d95d2bde9a887b7443

**Review by ribel — APPROVED** [review]
Changes were tested, issue is fixed and there are no regressions.

**Pull request merged**